src/view.c

```diff
@@ -273,6 +273,19 @@ set_adaptive_sync_fullscreen(struct view *view)
 	wlr_output_commit(view->output->wlr_output);
 }
 
+static void
+set_tearing(struct view *view)
+{
+	if (rc.allow_tearing < LAB_TEARING_ENABLED) {
+		return;
+	}
+	if (rc.allow_tearing == LAB_TEARING_FULLSCREEN && view->fullscreen) {
+		view->output->tearing = true;
+	} else {
+		view->output->tearing = view->tearing_hint;
+	}
```

Comment:
Thinking about this more, it might be more obvious if this function explicitly sets the tearing state in all cases, rather than relying on just preserving state for the disabled and always cases. For example:

```c
switch (rc.allow_tearing) {
case LAB_TEARING_DISABLED:
    view->output->tearing = false;
    break;
case LAB_TEARING_ALWAYS:
    view->output->tearing = true;
    break;
case LAB_TEARING_FULLSCREEN:
    view->output->tearing = view->fullscreen;
    break;
case LAB_TEARING_ENABLED:
    view->output->tearing - view->tearing_hint;
    break;
default:
    assert(false);
}
```

Reply:
It was done like that to use this function to set tearing only when its switched on or off depending on view, while disabled and always have tearing constantly off or on. I changed always to -1 to keep code shorter here. But yea switch case sounds like would make this more clear.